### PR TITLE
Dropped the Signature proto and replaced it with SignalAgglomUpdates.…

### PR DESCRIPTION
**Pull request by HatchingEgg**
**Dropped the Signature proto and replaced it with SignalAgglomUpdates.proto.**

Signature, formerly known as encrypted Unknown 6 payload was a name we came up with earlier. A few weeks has passed and all types and most fields of this protocol is known now. As we all know it acts as some kind of agglomerated package of all changed device and sensor information. That's the reason why SignalAgglomUpdates.proto was choosen here.

Besides this, all types and missing fields are available now. This protocol is dumped from the current 0.35 client. As noted in the protocol itself all types are confirmed except some int32 values that may be floats.

Have fun and prepare your MITM attacks to guess the missing values.

Your loved,
Buddy Team

### CODE REVIEW

**Comment by keyphact** [comment]
Still failing I'm afraid, are you able to fix?

**Comment by HatchingEgg** [comment]
Ok, now it works. Stupid Travis. :package:

**Comment by keyphact** [comment]
:ok_hand:

File: src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	bytes field22 = 22;  // replay check - Changes every 5 minutes or so. Generation unknown but pointed to by 0001B8614
+	uint64 epoch_timestamp_ms = 23;
+	repeated int64 request_hashes = 24;  // xxHash64 of the requests being sent with this agglom
+	uint64 field25 = 25;
+
+	// 100% - Reference iOS lib "LocationUpdate" structure for bridge
+	message LocationUpdate {
+		string name = 1;  // "network", "gps", "fused" - only these 3 values (com.nianticlabs.location.NianticLocationManager.class)
+		int64 timestamp_ms = 2;
+		double altitude = 4;
```
Comment (Grover-c13) [inline src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto]: I cant get the altitude, latitude and longitude to parse as double, when trying to load a decrypted signature into the proto, anyone else have similar issues?

File: src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+// I can't confirm them being floats or not. All others are confirmed.
+// Note2: In the native lib, it does not look like they actually serialize to float. They use float datatypes in native structures, but serialize to doubles
+// Note3: We really shouldn't call this 'Signature' it's an agglomerated update packet for device reading changes + request hashes
+message SignalAgglomUpdates {
+	repeated UnknownMessage field1 = 1;
+	int64 timestamp_ms_since_start = 2;
+	string field3 = 3;
+	repeated LocationUpdate location_updates = 4;  // Multiple location updates at a time. This is all the updates since the last time we sent a request
+	repeated AndroidGpsInfo android_gps_info = 5;
+	repeated UnknownMessage field6 = 6;
```
Comment (Grover-c13) [inline src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto]: Still possibly active status?

File: src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	repeated SensorUpdate sensor_updates = 7;  // All the sensor updates since the last time we sent a request. (Seems to actually be throttled to 1-3 at a time)
+	DeviceInfo device_info = 8;                // device info - need to find this still to verify everything
+	IOSDeviceInfo ios_device_info = 9;         // iOS only - likely device capabilities? (Or even simpler being iOS device version flags)
+	int32 location_hash_by_token_seed = 10;    // Hashed location using the auth token as the seed (hashed auth token -> location hash seed)
```
Comment (Grover-c13) [inline src/POGOProtos/Networking/Envelopes/SignalAgglomUpdates.proto]: name can problem be simplified IMO

**Pull request merged**